addons/website_sale_stock/controllers/main.py

```diff
@@ -26,3 +27,23 @@ def shop_payment_transaction(self, *args, **kwargs):
         if values:
             raise ValidationError(' '.join(values))
         return super().shop_payment_transaction(*args, **kwargs)
+
+
+class WebsiteSale(website_sale_controller.WebsiteSale):
+    @http.route(['/shop/add/stock_notification'], type="json", auth="public", website=True)
+    def add_stock_email_notification(self, email, product_id):
+        if not email_re.match(email):
+            return {'error': BadRequest("Invalid Email")}
```

Comment:
Why not raise the exception?
Also the text needs to be translated
```suggestion
        if not email_re.match(email):
            raise BadRequest(_("Invalid Email"))
```